app/cluster-api-provider-sidero/controllers/metalmachine_controller.go

```diff
@@ -363,6 +363,7 @@ func (r *MetalMachineReconciler) patchProviderID(ctx context.Context, cluster *c
 // patchServerInUse updates a server to mark it as "in use".
 func (r *MetalMachineReconciler) patchServerInUse(ctx context.Context, serverClass *metalv1alpha1.ServerClass, serverObj *metalv1alpha1.Server) error {
 	serverObj.Status.InUse = true
+	serverObj.Status.Clean = false
```

Comment:
I'm not sure this is the best place for this. We don't really have a great way of knowing that an installed has taken place. That is why I am kind of torn on having this status. Maybe I am missing something still?

Reply:
I think it's okay to over-wipe, but not okay to under-wipe.

We need to make sure though that we don't have nodes booted from the environment before server becomes `InUse`. So we should make sure iPXE doesn't deliver any image to the server (except for discovery one) until server is `InUse`.